src/vanilla/utils/splitAtom.ts

```diff
@@ -37,6 +37,10 @@ type SplitAtomAction<Item> =
       atom: PrimitiveAtom<Item>
       before?: PrimitiveAtom<Item>
     }
+  | {
+      type: 'bulk'
```

Comment:
I'm not sure if this type is very common. Do you have any alternative ideas?

Reply:
`reorder` might be better for my use case. But there is difference between `reorder` and `bulk`.
`bulk` might have removed or inserted items with or without reordering. We should think `bulk` as new atoms not related to old ones.
Should we consider `bulk` too as a separate type?

Reply:
yeah, "reorder" sounds something different. For me "bulk" doesn't sound like a verb. "bulk-change", too long.
If you don't have any other idea, we can use `bulk`, or maybe `unstable_bulk` would be better for now.

Reply:
Are you saying, we do not need `reorder` and only `bulk?` will suffice? For the naming i will come out something proper later.

Reply:
Yeah, it can cover the reorder use case, right?

Reply:
Yes

Reply:
[About the naming ](https://react.dev/learn/updating-arrays-in-state#updating-arrays-without-mutation)

Possible operations on an array i can think of are:
- insert (Insert single item. Do not considers multiples.)
- remove (Remove single item. Do not considers multiples.)
- replace (Whole new array)
- reorder (This is a special case of the `replace`. Without `insert` and `remove`)
- move (Move single item. This is a special case of the `reorder`. `reorder` is equivalent of multiple `move`)

Reply:
insert, remove, move are atomic operations (for single item only). And for other cases `replace` would be sufficient.
Is it correct?

Reply:
thanks for the summary. it made me clear that, _for now_ I would like to avoid `replace` (bulk updates) and only provide single operations.
For `replace`, I would mutate the original atom. I am happy to learn the pitfall with this approach.

Reply:
Okay, thank you.
I will close the PR for now.